datacite/client.py

```diff
@@ -130,44 +113,44 @@ def metadata_post(self, metadata):
         """
         headers = {'Content-Type': 'application/xml;charset=UTF-8', }
 
-        r = self._request_factory()
-        r.post("metadata", body=metadata, headers=headers)
+        request = self._create_request()
+        resp = request.post("metadata", body=metadata, headers=headers)
 
-        if r.code == 201:
-            return r.data
+        if resp.status_code == 201:
+            return resp.text
         else:
-            raise DataCiteError.factory(r.code, r.data)
+            raise DataCiteError.factory(resp.status_code, resp.text)
 
     def metadata_delete(self, doi):
         """Mark as 'inactive' the metadata set of a DOI resource.
 
         :param doi: DOI name of the resource.
         :return: "OK"
         """
-        r = self._request_factory()
-        r.delete("metadata/" + doi)
+        request = self._create_request()
+        resp = request.delete("metadata/" + doi)
 
-        if r.code == 200:
-            return r.data
+        if resp.status_code == 200:
+            return resp.text
         else:
-            raise DataCiteError.factory(r.code, r.data)
+            raise DataCiteError.factory(resp.status_code, resp.text)
 
     def media_get(self, doi):
         """Get list of pairs of media type and URLs associated with a DOI.
 
         :param doi: DOI name of the resource.
         """
-        r = self._request_factory()
-        r.get("media/" + doi)
+        request = self._create_request()
+        resp = request.get("media/" + doi)
 
-        if r.code == 200:
+        if resp.status_code == 200:
             values = {}
-            for line in r.data.splitlines():
+            for line in resp.text.splitlines():
```

Comment:
self-learn question: What is the reason to prefer `text` over `data`? I see that `text` is a lower level attribute not a main interface one. Ref [requests](https://docs.python-requests.org/en/master/api/#requests.Response.text). Is it to *make sure* to get unicode data and not possible failures if `data` returns bytes or something different than text?

Reply:
`text` is the property to get the response if I am not mistaken, see [here](https://docs.python-requests.org/en/latest/user/quickstart/#response-content). `data` does not exist in `requests`, it was wrapped before in the Client class.
No idea why in the doc is considered "Low level", that is the Response class.